File: scripts/dev_ocr_compare.py
#!/usr/bin/env python3
"""Optional Mac helper: run Tesseract CLI on an image (same role as dev_ocr_compare.rb)."""

from __future__ import annotations

import shutil
import subprocess
import sys
import tempfile
from pathlib import Path


def main() -> None:
    if len(sys.argv) < 2:
        print(f"Usage: {sys.argv[0]} <image.png>", file=sys.stderr)
        sys.exit(1)
    path = Path(sys.argv[1])
    if not path.is_file():
        print("File not found:", path, file=sys.stderr)
        sys.exit(1)
    ts = shutil.which("tesseract")
    if not ts:
        print(
            "tesseract not in PATH. Install: brew install tesseract\n"
            "The iOS app uses Apple Vision; this is only for desktop comparison.",
            file=sys.stderr,
        )
        sys.exit(1)
    with tempfile.TemporaryDirectory(prefix="ocr_compare_") as d:
        out = Path(d) / "out"
        subprocess.run([ts, str(path), str(out), "-l", "eng"], check=True)
        print((out.with_suffix(".txt")).read_text())


if __name__ == "__main__":
    main()

File: scripts/dev_ocr_compare.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Optional Mac developer helper: run Tesseract (C++ engine) on a scan for comparison
# with on-device Apple Vision OCR. Does not ship in the iOS app.
#
# Usage: ruby scripts/dev_ocr_compare.rb path/to/scan.png
# Requires: brew install tesseract (optional)

path = ARGV[0]
abort "Usage: #{$PROGRAM_NAME} <image.png>" unless path && File.file?(path)

def which(cmd)
  ex = ENV["PATH"].split(File::PATH_SEPARATOR).map { |p| File.join(p, cmd) }.find { |f| File.executable?(f) }
  ex || ""
end

ts = which("tesseract")
if ts.empty?
  warn "tesseract not found in PATH. Install with: brew install tesseract"
  warn "The iOS app uses Apple Vision (native); this script is only for desktop comparison."
  exit 1
end

require "tmpdir"
Dir.mktmpdir("ocr_compare") do |dir|
  txt = File.join(dir, "out")
  ok = system(ts, path, txt, "-l", "eng")
  abort "tesseract failed" unless ok
  puts File.read("#{txt}.txt")
end
